### ISSUE TITLE
Parameter for KeyPair Failure

### ISSUE THREAD

**Issue by darkn3rd** [aws-quickstart/taskcat#546]
### Description

Parameter for KeyPair is currently not working.

#### CONFIG

```yaml
# .taskcat.yml
project:
  name: dgraph-aws-marketplace
  regions:
  - us-west-1
  - us-west-2
tests:
  default:
    parameters:
      KeyName: dgraph-marketplace-cf-stack-key
    template: ./dgraph.json
```

#### COMMAND

```bash
taskcat test run
```

#### CFN TEMPLATE

The applicable parameter in question.

```json
{
    "Parameters": {
        "KeyName": {
            "Description": "Name of an existing EC2 KeyPair to enable SSH access to the instances",
            "Type": "AWS::EC2::KeyPair::KeyName",
            "ConstraintDescription": "must be the name of an existing EC2 KeyPair."
        }
}
```

Source Code for Template:

* https://gist.github.com/darkn3rd/4e482dbfd9748df3e95ebafa0875fddf

#### INSTALL STEPS

AWS profile used, working.

```bash
brew install pyenv 
brew install pyenv-virtualenv
pyenv install 3.8.0
echo 'aws' > .python-version 
pyenv virtualenv 3.8.0 aws
cd .
pip install taskcat
pip install awscli
```

### ACTUAL BEHAVIOR

```
[INFO   ] : Parameter validation failed: parameter value dgraph-marketplace-cf-stack-key for
parameter name KeyName does not exist
....
[ERROR  ] : One or more stacks failed tests: {'arn:aws:cloudformation:us-west-1:726426114003:stack/tCaT-dgraph-aws-marketplace-default-3df9cd826d8e40798af8099e93547c0a/493a48c0-80d4-11ea-b634-06093e2fdba5': 'Parameter validation failed: parameter value dgraph-marketplace-cf-stack-key for parameter name KeyName does not exist', 'arn:aws:cloudformation:us-west-2:726426114003:stack/tCaT-dgraph-aws-marketplace-default-3df9cd826d8e40798af8099e93547c0a/495327f0-80d4-11ea-ab48-026696022230': 'Parameter validation failed: parameter value dgraph-marketplace-cf-stack-key for parameter name KeyName does not exist'}
```

### EXPECTED BEHAVIOR

I expected that this would work, as I can run it fine with aws cli, but only fails with taskcat:

```bash
region=$(aws configure get region) # us-west-2
s3_bucket_name=some-random-whatever-bucket
stack_name=dgraph-marketplace-cf-stack-stack
ssh_key_name=dgraph-marketplace-cf-stack-key

aws s3 ls --region ${region} "s3://${s3_bucket_name}"
aws cloudformation deploy \
 --capabilities CAPABILITY_IAM \
 --template-file dgraph.json \
 --s3-bucket "${s3_bucket_name}" \
 --stack-name "${stack_name}" \
 --region "${region}" \
 --parameter-overrides \
  KeyName="${ssh_key_name}"
```

### VERSIONS

* TaskCat `0.9.17`

**Comment by andrew-glenn** [comment]
Low hanging fruit...

Iwant to verify that the region you have in the `$region` variable is what you expect. Also please confirm that there are no `auth` entries in your `~/.taskcat.yml` (global). 

We leverage this concept daily and run into what you’re experiencing when what we expect isn’t what actually is present.

**Comment by darkn3rd** [comment]
That `.taskcat.yml` is pretty much it, I don't have a global specified. I made one edition for experimentation, in putting the parameters under `project`.  Is there a way to print the final rendered config (plus defaults)?  When you say `$region`, you mean the entries under `regions:` or something else?  These are indeed expected under `regions:`.  I wanted to test in 2 AZ and 3 AZ specifically, as the CFN script was coded to responds to either.  When this works, I will test in 16 regions.

**Comment by andrew-glenn** [comment]
I'm looking to verify via the AWSCLI that the keypair `dgraph-marketplace-cf-stack-key` exists in `us-west-1` and `us-west-2`. 

Both commands below should return a value.
```
aws ec2 describe-key-pairs --key-names dgraph-marketplace-cf-stack-key --region us-west-1
aws ec2 describe-key-pairs --key-names dgraph-marketplace-cf-stack-key --region us-west-2
```

**Comment by darkn3rd** [comment]
```
aws ec2 describe-key-pairs --key-names dgraph-marketplace-cf-stack-key --region us-west-1
{
    "KeyPairs": [
        {
            "KeyPairId": "key-0e550c45941cdc9a0",
            "KeyFingerprint": "b8:ab:da:53:bd:33:99:27:ef:71:d4:52:a3:a9:4e:b9",
            "KeyName": "dgraph-marketplace-cf-stack-key",
            "Tags": []
        }
    ]
}

aws ec2 describe-key-pairs --key-names dgraph-marketplace-cf-stack-key --region us-west-2
{
    "KeyPairs": [
        {
            "KeyPairId": "key-0a1cb4ee019128330",
            "KeyFingerprint": "b8:ab:da:53:bd:33:99:27:ef:71:d4:52:a3:a9:4e:b9",
            "KeyName": "dgraph-marketplace-cf-stack-key",
            "Tags": []
        }
    ]
}
```

**Comment by andrew-glenn** [comment]
@darkn3rd  - sorry for the delay. This is odd. I'm not necessarily able to replicate. I know it seems cliche, but we just released 0.9.18 and it included a number of bugfixes. Would you mind upgrading to 0.9.18 and verifying this is still present? I'm happy to dive in further.

**Comment by darkn3rd** [comment]
@andrew-glenn Really?  It reproduces 100%.  I can try the new version.  

I have the full source code (in case anything obvious I missed) in public domain: 
* https://github.com/dgraph-io/dgraph/tree/master/contrib/config/marketplace/aws

**Comment by andrew-glenn** [comment]
@darkn3rd - I just cloned the repo and worked directly from what you have there (on a fresh laptop) , including seeding the public keys. I was able to build the `dgraph` stack in most regions (exception being a couple EU regions due to an unsupported configuration). 

In the process I also copied the .taskcat.yml file from `tests/` up one directory to `aws` before running the tests, and it picked up the template without issue on `0.9.17`.
